File: engine/src/main/java/io/github/aemogie/timble/engine/utils/Debug.java
package io.github.aemogie.timble.engine.utils;

import io.github.aemogie.timble.gl.window.Window;
import io.github.aemogie.timble.utils.events.Event.Listener;

import static org.lwjgl.opengl.GL11.*;

public class Debug {
	public static final Listener<Window.FrameLoopEvent> TRIANGLE = new Listener<>(false) {
		@Override
		protected boolean onFire(Window.FrameLoopEvent event) {
			glBegin(GL_TRIANGLES);
			glVertex2f(-0.5f, -0.5f);
			glVertex2f(0.0f, 0.5f);
			glVertex2f(0.5f, -0.5f);
			glEnd();
			return true;
		}
	};
}

File: gl/src/main/java/io/github/aemogie/timble/gl/utils/Debug.kt
package io.github.aemogie.timble.gl.utils

import org.lwjgl.opengl.GL11.*

fun legacyTriangle() {
	glClearColor(0f, 0f, 0f, 1f)
	glBegin(GL_TRIANGLES)
	glVertex2f(-0.5f, -0.5f)
	glVertex2f(0.0f, 0.5f)
	glVertex2f(0.5f, -0.5f)
	glEnd()
}